src/DotVVM.Core/Controls/IPageableGridViewDataSet.cs

```diff
@@ -1,73 +1,39 @@
-using System;
-using System.Threading.Tasks;
-
 namespace DotVVM.Framework.Controls
 {
     /// <summary>
-    /// Extends the base GridViewDataSet with paging functionality.
+    /// Extends the <see cref="IBaseGridViewDataSet" /> with paging functionality.
     /// </summary>
     public interface IPageableGridViewDataSet : IBaseGridViewDataSet
     {
         /// <summary>
-        /// Gets or sets an object that represents the settings for paging.
+        /// Gets the settings for paging.
         /// </summary>
-        IPagingOptions PagingOptions { get; set; }
+        IPagingOptions PagingOptions { get; }
 
         /// <summary>
         /// Navigates to the first page.
         /// </summary>
-        [Obsolete("Use GoToFirstPageAsync")]
         void GoToFirstPage();
 
-        /// <summary>
-        /// Navigates to the first page.
-        /// </summary>
-        /// <returns></returns>
-        Task GoToFirstPageAsync();
-
         /// <summary>
         /// Navigates to the last page.
         /// </summary>
-        [Obsolete("Use GoToFirstLastAsync")]
         void GoToLastPage();
 
         /// <summary>
-        /// Navigates to the last page.
+        /// Navigates to the next page if possible.
         /// </summary>
-        Task GoToLastPageAsync();
-
-        /// <summary>
-        /// Navigates to the next page (if possible).
-        /// </summary>
-        [Obsolete("Use GoToNextPageAsync")]
         void GoToNextPage();
```

Comment:
Do we need these helpers, when you can get the page from `PagingOptions` and then call `GoToPage`? What about to make this an extension method?

Reply:
Yes, we should keep these helpers. But they can be extension methods.